tuf/api/metadata.py

```diff
@@ -1153,23 +1154,51 @@ def from_dict(cls, delegations_dict: Dict[str, Any]) -> "Delegations":
         for keyid, key_dict in keys.items():
             keys_res[keyid] = Key.from_dict(keyid, key_dict)
         roles = delegations_dict.pop("roles")
-        roles_res = []
+        roles_res: "OrderedDict[str, DelegatedRole]" = OrderedDict()
         for role_dict in roles:
             new_role = DelegatedRole.from_dict(role_dict)
-            roles_res.append(new_role)
+            role_name = new_role.name
+            if roles_res.get(role_name) is not None:
+                raise ValueError(
+                    f"Delegated role names should be unique! "
+                    f"Got a role name '{role_name}' which is duplicate!"
+                )
```

Comment:
I'm confident you could squeeze the error into a single line

Creating temporary variables like role_name is not a bad practice when the variable makes things more readable or prevents multiple dictionary lookups... I'm not sure there is a reason in this case?

Reply:
Addressed both the temporary variable and made the comment one line.